File: atrs-view/src/src/app/atrs/cruds/project.service.js
(function() {
    'use strict';

    angular
        .module('atrs-module')
        .factory('projectService', ProjectService);
        

    /* @ngInject */
    function ProjectService($http, API_CONFIG) {
    	var urlBase = API_CONFIG.restAppName+'/project' ;
        var service = {
            activate: activate,
        	archive: archive,
            save: save,
            load: load,
            remove: remove,
            find: find,
            list: list
        };
        return service;
        
        function save(project) {
        	return $http.post(urlBase+'/save', project);
        };
        
        function archive(project) {
        	return $http.post(urlBase+'/archive', project);
        };
        
        function activate(project) {
        	return $http.post(urlBase+'/activate', project);
        };

        function load (id) {
        	return $http.put(urlBase+'/load/' + id);
        };

        function remove (id) {
            return $http.delete(urlBase + '/delete/' + id);
        };

        function list () {
            return $http.get(urlBase + '/list');
        };

        function find (searchObject) {
            return $http.post(urlBase + '/find', searchObject);
        };
    }
})();

File: atrs-service/src/main/java/com/crossover/atrs/service/ProjectService.java
package com.crossover.atrs.service;

import java.util.Date;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Sort;
import org.springframework.data.domain.Sort.Direction;
import org.springframework.data.domain.Sort.Order;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.stereotype.Service;

import com.crossover.atrs.model.Project;
import com.crossover.atrs.respository.ProjectRepository;
import com.crossover.atrs.to.ProjectSearchTo;

@Service
public class ProjectService {

	@Autowired
	private ProjectRepository repository;
	
	@Autowired
	private MongoTemplate mongoTemplate;

	public Project load(String id) {
		return this.getRepository().findOne(id);
	}
	
	public Project save(Project project) {
		return this.getRepository().save(project);
	}
	
	public Page<Project> search(ProjectSearchTo searchTo) {
		if(searchTo == null) {
			return null;
		}		
		if(searchTo.getSearchTerm() == null) {
			// TODO [valverde.thiago] Retirar essa camanga daqui. Query da pau com termos nulos
			searchTo.setSearchTerm("");
		}
		if(searchTo.getSort() == null) {
			searchTo.setSort(new Sort(new Order(Direction.ASC, "title")));
		}
		return this.getRepository().search(searchTo.getSearchTerm(), searchTo.isOnlyActives(), searchTo);
	}

	public Iterable<Project> listAll() {
		return this.getRepository().findAll();
	}

	public Project archive(Project project) {
		if(project == null || project.getArchivationDate() != null) {
			return project;
		}
		project.setArchivationDate(new Date());
		return this.save(project);
	}

	public Project activate(Project project) {
		if(project == null || project.getArchivationDate() == null) {
			return project;
		}
		project.setArchivationDate(null);
		return this.save(project);
	}
	
	protected ProjectRepository getRepository() {
		return repository;
	}

	protected MongoTemplate getMongoTemplate() {
		return mongoTemplate;
	}
	
}